### DISCUSSION TITLE
Exclude keywords

### DISCUSSION THREAD

**Discussion by lelandstarr** [modelint/scrall#5] [Ideas]
https://github.com/modelint/scrall/blob/8b9dd84019e1e23ec8e7c0cf04cf64224901f89e/Grammar/scrall.peg#L23

```
//ignore_keywords = (
//    !" true" !"true " !" false" !"false "
//    !" and" !"and " !" or" !"or "!" not" !"not "
//    !" its" !"its " ! " me" !" me"
//)
word = r'[A-Za-z][A-Za-z0-9]*'
//word = ignore_keywords r'[A-Za-z][A-Za-z0-9]*'
name = word (NAME_GLUE word)* // One word or a sequence of delimited words (name used for most model elements)
```

if you attach the negative lookahead to WORD rather then name, it will catch the keywords AS words.  This way you do not have to worry about white space.

I have this working in java and have checked that python also has the same negative lookahead syntax. 

```
//Here is word as allowed_word with the keyword exclusion.
allowed_word = r'(?!(true|false|and|or|me|its|not))[A-Za-z][A-Za-z0-9]*'

// and then name would still be
name = allowedword (NAME_GLUE allowedword)* 

```

**Comment by modelint** [comment]
Thanks!  Yeah, I knew there was trouble there and wasn't super happy with my 'solution'
